connection.go

```diff
@@ -244,8 +244,28 @@ func (c *connection) sendMessageToServer(guid string, method string, params inte
 	if c.tracingCount.Load() > 0 && len(stack) > 0 && guid != "localUtils" {
 		c.LocalUtils().AddStackToTracingNoReply(id, stack)
 	}
-	result := <-cb.(chan callback)
```

Comment:
why was this change needed?

Reply:
I did encounter a situation where the server did not return a timeout message even though the timeout was passed when calling `route.fetch`, resulting in a deadlock of the route. Similar to the issue mentioned in #362. This is uncommon in end-to-end testing and is more likely to occur in scraping.

But the root cause was not found because it was difficult to reproduce. In order to prevent the client from hanging, add this part of the modification.

Reply:
I would prefer to not landing this, since then we do something different to what Python/Java/Go/Node is doing.

https://github.com/playwright-community/playwright-go/issues/362#issuecomment-1655801180 is also filed upstream at e.g. https://github.com/microsoft/playwright/issues/22590 which we fixed last week.

Reply:
Ok, thanks. I rolled back the changes. `b.Lock()` in browser_context.go is redundant and can lead to deadlocks, it's kept.